desafios/bundleDesafios_solucoes/reactJs-GabrielWolf-Dev/src/Components/SecondChallenge/index.js

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from 'react';
+import Header from '../Header';
+import './chess.css';
+
+export default function SecondChallenge() {
+    const [matrix, setMatrix] = useState([
+        "0 0 0 0 0 0 0 0",
+        "0 0 0 0 0 0 0 0",
+        "0 0 0 0 0 0 0 0",
+        "0 0 0 1 1 0 0 0",
+        "0 0 0 1 1 0 0 0",
+        "0 0 0 0 0 0 0 0",
+        "4 0 0 0 0 0 0 4",
+        "0 0 0 0 0 6 0 0",
+    ]);
+    const objPieces = {
+        peao: 0,
+        bispo: 0,
+        cavalo: 0,
+        torre: 0,
+        rainha: 0,
+        rei: 0,
+    };
+    const [pieces, setPieces] = useState(objPieces);
+
+    useEffect(() => {
+        matrix.forEach(matrix => {
+            const row = matrix.split(' ');
+
+            row.forEach(row => {
+                if(row === '1')
+                    setPieces({ ...pieces, peao: pieces.peao += 1 });
+
+                if(row === '2')
+                    setPieces({ ...pieces, bispo: pieces.bispo += 1 });
+
+                if(row === '3')
+                    setPieces({ ...pieces, cavalo: pieces.cavalo += 1 });
+
+                if(row === '4')
+                    setPieces({ ...pieces, torre: pieces.torre += 1 });
+
+                if(row === '5')
+                    setPieces({ ...pieces, rainha: pieces.rainha += 1 });
+
+                if(row === '6')
+                    setPieces({ ...pieces, rei: pieces.rei += 1 });
```

Comment:
```suggestion
                const piecesDictionary = {
                    1: 'peao',
                    2: 'bispo',
                    3: 'cavalo',
                    4: 'torre',
                    5: 'rainha',
                    6: 'rei'
                }
                
                const piece = piecesDictionary[row]
                
                setPieces({...pieces, [piece]: pieces[piece] += 1})
```

Eu acho que desse jeito ficaria menos verborrágico.